compiler/optimizer/src/main/java/edu/snu/nemo/compiler/optimizer/policy/PolicyImpl.java

```diff
@@ -66,7 +68,8 @@ private CompiletimeOptimizer() {
     if (passes.hasNext()) {
       final CompileTimePass passToApply = passes.next();
       // Apply the pass to the DAG.
-      final DAG<IRVertex, IREdge> processedDAG = passToApply.apply(dag);
+      final DAG<IRVertex, IREdge> processedDAG = passToApply.getCondition().test(dag)
+          ? passToApply.apply(dag) : dag;
```

Comment:
I'd skip integrity checking and `processedDAG.storeJSON`, when the pass does not run due unmet conditions.
How about also logging this information to let ourselves know which pass ran and did not ran?

e.g.,
INFO: Pass X ran (JSON stored at ABC)
INFO: Pass Y did not ran due to condition C1
INFO: Pass Z ran (JSON stored at DEF)

Reply:
Good idea! I'll apply that on the code.